src/ng/directive/ngOptions.js

```diff
@@ -534,11 +565,6 @@ var ngOptionsDirective = ['$compile', '$parse', function($compile, $parse) {
       // We will re-render the option elements if the option values or labels change
       scope.$watchCollection(ngOptions.getWatchables, updateOptions);
 
-      // We also need to watch to see if the internals of the model changes, since
-      // ngModel only watches for object identity change
-      if (ngOptions.trackBy) {
-        scope.$watchCollection(attr.ngModel, function() { ngModelCtrl.$render(); });
-      }
```

Comment:
Using $watchCollection on the model was misguided.
This would:
* unnecessarily watch all the properties on a non-multiple select's model
* not react to changes if the tracked value was based on a deeper property in the model on a non-multiple select
* not react to changes in a multple select if only the tracked property value was changed on an item in the model collection.